Core/Src/main.c

```diff
@@ -86,15 +93,30 @@ int main(void)
 
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
-  MX_USART2_UART_Init();
-  /* USER CODE BEGIN 2 */
+  MX_SPI1_Init();
+  MX_TIM1_Init();
 
+  /* USER CODE BEGIN 2 */
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_6, GPIO_PIN_SET); // pull CS high
+  HAL_TIM_PWM_Start(&htim1, TIM1_CH1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_6, GPIO_PIN_RESET); // pull CS low
+	  HAL_SPI_TransmitReceive(&hspi, *spi_buf_tx, *spi_buf_rx, Size, Timeout);
```

Comment:
The `spi_buf_tx` and `spi_buf_rx` is already pointers since they are declared as arrays. `*spi_buf_tx` is saying the take the unreferenced value of that pointer, and that makes it just a variable.